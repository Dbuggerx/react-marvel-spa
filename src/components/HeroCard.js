// @flow

import React from 'react';
import type { Hero } from '../services/types';

type Props = {
    hero: Hero,
    imageUrl: string,
};

export default (props: Props) => (
    <div>
        <img src={props.imageUrl} alt={props.hero.name} />
        Name: {props.hero.name}
    </div>
);
