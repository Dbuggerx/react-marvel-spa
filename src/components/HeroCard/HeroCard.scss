@import '../scss/variables';

.hero-card {
  background: map-get($colors, highlight-bg);
  color: map-get($colors, highlight-fg);
  text-align: center;
  padding: 5px;
  height: 100%;
  width: 100%;
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  transition: transform 0.3s ease-out;
  backface-visibility: hidden;
  transform: translateZ(0);
  -webkit-font-smoothing: subpixel-antialiased;

  &:focus {
    outline-color: map-get($colors, highlight-bg);
  }

  &:hover, &:focus {
    transform: scale(1.1);
    box-shadow: 0px 0px 12px 3px rgba(0,0,0,0.75);
    z-index: 2;
  }

  &__label {
    max-width: 100%
  }
}