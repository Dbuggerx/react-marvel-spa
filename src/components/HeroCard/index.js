// @flow

import React, { Component } from 'react';
import type { Hero } from '../../services/types';
import './HeroCard.scss';

type Props = {
    hero: Hero,
    imageUrl: string,
    onClick: (hero: Hero) => void,
};

/**
 * Renders a card that displays the passed Hero data and its image
 */
export default class HeroCard extends Component<Props> {
    handleClick = () => {
        this.props.onClick(this.props.hero);
    };

    handleKeyUp = (event: SyntheticKeyboardEvent<*>) => {
        if ('key' in event ? event.key === 'Enter' : event.keyCode === 13) this.props.onClick(this.props.hero);
    };

    render() {
        return (
            <div className="hero-card" onClick={this.handleClick} onKeyUp={this.handleKeyUp} role="button" tabIndex="0">
                <img src={this.props.imageUrl} alt={this.props.hero.name} />
                <div className="hero-card__label">{this.props.hero.name}</div>
            </div>
        );
    }
}
