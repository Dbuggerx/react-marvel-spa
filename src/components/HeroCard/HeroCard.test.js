import React from 'react';
import { shallow } from 'enzyme';
import HeroCard from './index';

const basicHeroMock = {
    id: 1011334,
    name: '3-D Man',
    description: 'This is a test!',
    modified: '2014-04-29T14:18:17-0400',
    thumbnail: {
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
        extension: 'jpg',
    },
};

describe('HeroCard', () => {
    test('renders properly', () => {
        const wrapper = shallow(<HeroCard hero={basicHeroMock} imageUrl="the/img/url" />);
        expect(wrapper).toMatchSnapshot();
    });


    test('it calls props.onClick upon click', () => {
        const handleClickMock = jest.fn();
        const wrapper = shallow(<HeroCard hero={basicHeroMock} imageUrl="the/img/url" onClick={handleClickMock} />);
        wrapper.find('[role="button"]').simulate('click');
        expect(handleClickMock).toHaveBeenCalledTimes(1);
    });

    test('it calls props.onClick upon keyup', () => {
        const handleClickMock = jest.fn();
        const wrapper = shallow(<HeroCard hero={basicHeroMock} imageUrl="the/img/url" onClick={handleClickMock} />);
        const target = wrapper.find('[role="button"]');
        target.simulate('keyUp', {
            key: 'Enter',
        });
        target.simulate('keyUp', {
            keyCode: 13,
        });
        target.simulate('keyUp', {
            key: 'wrongOne',
        });
        expect(handleClickMock).toHaveBeenCalledTimes(2);
    });
});
