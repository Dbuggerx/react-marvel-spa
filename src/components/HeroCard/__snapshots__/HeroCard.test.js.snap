// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard renders properly 1`] = `
<div
  className="hero-card"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={
    Object {
      "backgroundImage": "url(the/img/url)",
    }
  }
  tabIndex="0"
>
  <div
    className="hero-card__label"
  >
    3-D Man
  </div>
</div>
`;
